### PR TITLE
Check a mesh resolution

### PR DESCRIPTION
**Pull request by olegrog**
Check if J_MX provides a sufficient resolution for solid--liquid interface thickness.

### CODE REVIEW

File: slm_porosity/heat_transfer.f90 [hunk 1]
```diff
@@ -755,26 +763,43 @@ SUBROUTINE user_pre_process
...
-    for_du = RESHAPE((/ h_star /), SHAPE(for_du))
-    CALL c_diff_fast(for_du, du, du_dummy, j_lev, nwlt, grad_meth(meth), 10, n_integrated, 1, n_integrated)
-    CALL reallocate_vector(grad_h_star_prev); grad_h_star_prev = du(1,:,:)
+
+    ! 3. Check if J_MX provides a sufficient resolution for solid--liquid interface thickness
+    IF (check_resolution) THEN
+      max_grad_temp = MAXVAL(ABS(du(2,:,:)), 1)
+      CALL parallel_vector_sum(REALMAXVAL=max_grad_temp, LENGTH=dim)
+      thickness = interface_thickness/max_grad_temp
+      resolution = (xyzlimits(2,:) - xyzlimits(1,:)) / (mxyz*2**j_mx)
```
Comment (shmidt9510) [inline slm_porosity/heat_transfer.f90:790]: Does this formula have any physical meaning?

Comment (olegrog) [reply slm_porosity/heat_transfer.f90:790]: `thickness` has a physical meaning. We want to resolve it and, therefore, to compare with the minimum mesh step.

File: slm_porosity/heat_transfer.f90 [hunk 2]
```diff
@@ -755,26 +763,43 @@ SUBROUTINE user_pre_process
...
+      IF (par_rank.EQ.0) THEN
+        WRITE(*, '(A35, 3ES10.2)') 'The minimum interface thickness =', thickness
+        WRITE(*, '(A35, 3ES10.2)') 'The minimum mesh step =', resolution
+        WRITE(*, '(A35, 3ES10.2)') 'Their ratio =', thickness/resolution
+      END IF
+      IF (ANY(thickness.LT.resolution*thickness_points)) STOP '--- Insufficient resolution ---'
```
Comment (shmidt9510) [inline slm_porosity/heat_transfer.f90:796]: Okey, then `Insufficient ` here have any additional math proof. And want to resolve phrase above? Maybe I missed some points yesterday.

Comment (olegrog) [reply slm_porosity/heat_transfer.f90:796]: Wavelets give us the universal instrument to control resolution just by analyzing the wavelet coefficients (mathematical indicator). But optionally we can look after the number points lying on the interface thickness (physical indicator).

**Pull request closed without merging**